src/api/code-challenge/content-types/code-challenge/test-evaluator.ts

```diff
@@ -0,0 +1,111 @@
+import {
+  evaluateWithContext,
+  getCode,
+  getEvaluationContext,
+  overrideConsoleLog,
+  restoreConsoleLog,
+} from "../code-challenge/utils";
+
+type runTestEvaluatorProps = {
+  metaCaseCode: string;
+  internalTestCode: string;
+  metaLabel: string;
+  challengeLabel: string;
+  metaTestId: number;
+  evalResultShouldBe: boolean;
+  internalTestId: number;
+  removeComments?: boolean;
+};
+
+export const runTestEvaluator = async ({
+  metaCaseCode,
+  internalTestCode,
+  metaTestId,
+  evalResultShouldBe,
+  internalTestId,
+  removeComments,
+}: runTestEvaluatorProps) => {
+  const getShortError = () => {
+    return `${evalError.stack
+      ?.toString()
+      .substring(0, 250)} --- (End of Abridged Error)`;
+  };
+
+  const getNoError = () => {
+    return "no eval() error detected";
+  };
+
+  const getMetaTestResultObject = () => {
+    switch (true) {
+      case evalResultShouldBe === false:
+        return {
+          evalError: evalError ? getShortError() : getNoError(),
+          userPassed: false,
+          description: `FAIL: Failing Examples currently not supported`,
+          evalResultType: typeof evalResult,
+          evalResult: evalResult,
+        };
+      case typeof evalResult === "string":
+        return {
+          evalError: evalError ? getShortError() : getNoError(),
+          userPassed: false,
+          description: `FAIL: EvalResultType should be 'boolean', but currently is 'string',\nSuggestion: Check for extra quotes around internal/metaCaseCodes`,
+          evalResultType: typeof evalResult,
+          evalResult: evalResult,
+        };
+      case typeof evalResult !== "boolean" && typeof evalResult !== "string":
+        return {
+          evalError: evalError ? getShortError() : getNoError(),
+          userPassed: false,
+          description: `FAIL: EvalResultType should always be 'boolean' but is currently '${typeof evalResult}'.`,
+          evalResultType: typeof evalResult,
+          evalResult: evalResult,
+        };
```

Comment:
I would honestly just put this as the last 'default' case but that might just be a matter of opinion

Reply:
I think that makes sense. I changed it.